viewing_party/party.py

```diff
@@ -1,23 +1,137 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if (not title) or (not genre) or (not rating):
+        return None
+      
+    if not all((isinstance(title, str), isinstance(genre, str),
+               isinstance(rating, float))):
+        return None
+
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    # Handles if create_movie() returns None for movie
+    if movie:
+        user_data["watched"].append(movie)
+       
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    if movie:
+        user_data["watchlist"].append(movie)
+   
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            move_movie = movie
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(move_movie)

```

Comment:
Careful here! We are removing items from a list _while_ iterating through it. This can result in items being skipped over because of a shift in index positions.

How best to combat this can depend. We could make a copy of the watchlist and remove from one while we loop through the other. Or perhaps we can `break` once we finish line 39. But then that begs the question, what if the same movie is inside the list?

Something to think about!